### PR TITLE
Task/03 components

### PR DESCRIPTION
**Pull request by roman-orlovskyi-moc**
Acceptance Criteria for 1
- [x] Mentor is added to pr
- [x] AC copied to pr description
- [x] Link to deployed version is added to pr
- [ ] The `ProductsList` component correctly receives product data as props.
- [x] `ProductCard` component created and renders a single product.
- [x] `Header` and `Footer` elements are present.
- [x] `ProductList` displays right on mobile devices

Acceptance Criteria for 1*
- [x] The `Header` component has clickable elements that toggle the display between the `About` and `ProductsList` components.
- [x] State management is handled in `App.tsx`, with state determining which component to show.
- [ ] `LocalStorage` added to save items in card after page realod

[deployed version](https://roman-orlovskyi-moc.github.io/fe-react-2024/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,21 +1,33 @@
...
+
     return (
-        <>
-            <HeaderComponent />
+        <AppContextProvider>
+            <HeaderComponent shouldShowAboutPage={shouldShowAboutPage} toggleShowAboutPageState={toggleShowAboutPage} />
             <main className="mainContentWrapper contentWrapper">
-                <AboutMeComponent fullName={FULL_NAME} nikName={NIK_NAME} />
+                {shouldShowAboutPage && <AboutComponent fullName={FULL_NAME} nikName={NIK_NAME} />}
+                {!shouldShowAboutPage && <ProductsListComponent />}
```
Comment (vladyslav-yermolin-moc) [inline src/App.tsx]: You can use here ternary operator
```suggestion
                {shouldShowAboutPage ?
                  <AboutComponent fullName={FULL_NAME} nikName={NIK_NAME} /> : 
                  <ProductsListComponent />}
```

File: src/components/common/Pagination.component.tsx [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    const setPreviousPage = (pageNumber: number) => setCurrentPage(pageNumber <= 1 ? 1 : pageNumber - 1);
+    const setNextPage = (pageNumber: number) => setCurrentPage(pageNumber >= totalPages ? totalPages : pageNumber + 1);
+
+    return (
+        <nav>
+            <ul className={styles.paginationList}>
+                <li>
+                    <button
+                        className={`${styles.paginationButton} ${styles.paginationArrowButton}`}
+                        onClick={setPreviousPage.bind(null, page)}
```
Comment (vladyslav-yermolin-moc) [inline src/components/common/Pagination.component.tsx]: as an optional ehancement
```suggestion
                        onClick={() => setPreviousPage(page)}
```

File: src/components/header/header.module.css [hunk 3]
```diff
@@ -1,6 +1,8 @@
 .header {
   width: 100%;
   padding: 17px 0;
+  color: #FFF;
+  background-color: #111;
 }
 
 .headerLogo {
```
Comment (vladyslav-yermolin-moc) [inline src/components/header/header.module.css]: Your header looks bigger than it has to be. As a suggestion, you can make a link flex container
image

File: src/components/common/pagination.module.css [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+.paginationButton:disabled {
+  cursor: not-allowed;
+}
+
+.paginationButton:disabled:hover {
+  background-color: var(--pagination-disabled-button-hover-background-color);
+  border-color: var(--pagination-disabled-button-border-color);
+}
+
+.paginationButtonActive {
```
Comment (vladyslav-yermolin-moc) [inline src/components/common/pagination.module.css]: Awesome pagination! And you've done theme-switching, amazing!
As small fixes: please correct the color of the active button for the light theme 
image

Comment (vladyslav-yermolin-moc) [reply src/components/common/pagination.module.css]: and buttons color for the dark theme
image

File: src/components/common/CartIconCounter.component.tsx
Comment (vladyslav-yermolin-moc) [inline src/components/common/CartIconCounter.component.tsx:1]: here I suggest putting CartIconCounter and Pagination in separate folders, together with their styles
```
CartIconCounter
  CartIconCounter.component
  cart-icon-counter.module
Pagination
  Pagination.component
  pagination.module.css
```

File: src/components/products-list/ProductsList.component.tsx [hunk 5]
```diff
@@ -0,0 +1,35 @@
+import { useState } from 'react';
+
+import { PaginationComponent } from '@/components/common/Pagination.component.tsx';
+
+import { ProductCardComponent } from './ProductCard.component.tsx';
+import { ProductsDataProviderComponent } from './ProductsDataProvider.component.tsx';
+
+import styles from './products-list.module.css';
+
+export const ProductsListComponent = () => {
```
Comment (vladyslav-yermolin-moc) [inline src/components/products-list/ProductsList.component.tsx]: Actually, we have the AC, where is explicitly stated
> The ProductsList component correctly receives product data **as props**.

But as you like to go further and get it from the context, that is OK

**Review by vladyslav-yermolin-moc — CHANGES_REQUESTED** [review]
Brilliant work! Please fix the minor design related remarks, and we are ready to go further

File: src/App.tsx (hunk 1 above)
Comment (roman-orlovskyi-moc) [reply src/App.tsx]: fixed

File: src/components/common/Pagination.component.tsx (hunk 2 above)
Comment (roman-orlovskyi-moc) [reply src/components/common/Pagination.component.tsx]: fixed

File: src/components/header/header.module.css (hunk 3 above)
Comment (roman-orlovskyi-moc) [reply src/components/header/header.module.css]: fixed

File: src/components/common/pagination.module.css (hunk 4 above)
Comment (roman-orlovskyi-moc) [reply src/components/common/pagination.module.css]: fixed

File: src/components/common/CartIconCounter.component.tsx
Comment (roman-orlovskyi-moc) [reply src/components/common/CartIconCounter.component.tsx:1]: fixed

File: src/components/products-list/ProductsList.component.tsx (hunk 5 above)
Comment (roman-orlovskyi-moc) [reply src/components/products-list/ProductsList.component.tsx]: fixed

File: src/components/common/pagination.module.css (hunk 4 above)
Comment (vladyslav-yermolin-moc) [reply src/components/common/pagination.module.css]: dark theme you can fix in the next pr

File: src/components/products-list/ProductsList.component.tsx (hunk 5 above)
Comment (vladyslav-yermolin-moc) [reply src/components/products-list/ProductsList.component.tsx]: As for me, now it looks much simpler. 
Context is an additional level of abstraction, which could help but also increase the perception difficulty.
Thanks for your fix.

**Review by vladyslav-yermolin-moc — APPROVED** [review]
As your main fixes were done yesterday, I'm gonna evaluate your work as `1+*`
Thanks

**Pull request merged**